dataland-frontend/tests/e2e/specs/eu-taxonomy-financials/DataIntegrity.ts

```diff
@@ -0,0 +1,95 @@
+        );
+        cy.get('div[name="onDemandInterbankLoans"]').should("not.exist");
+        cy.get('div[name="tradingPortfolio"]').should("not.exist");
+      });
+    });
+
+    it("Create a CreditInstitution (individual field submission)", () => {
+      createCompanyAndGetId("MTM Credit Institution").then((companyId) => {
+        cy.visitAndCheckAppMount(`/companies/${companyId}/frameworks/eutaxonomy-financials/upload`);
+        cy.get("select[name=financialServicesTypes]").select("Credit Institution");
+        cy.get("select[name=attestation]").select("Limited Assurance");
+        cy.get('input[name="reportingObligation"][value=Yes]').check();
+        cy.get("input[name=tradingPortfolio]").type("0.25");
+        submitEuTaxonomyFinancialsUploadFormAndGetDataId();
+        cy.visitAndCheckAppMount(`/companies/${companyId}/frameworks/eutaxonomy`);
+        cy.get('div[name="tradingPortfolio"]').should("contain", "Trading portfolio").should("contain", "25");
+        cy.get('div[name="onDemandInterbankLoans"]')
+          .should("contain", "On demand interbank loans")
+          .should("contain", "No data has been reported");
+        cy.get("body").should("not.contain", "Trading portfolio & on demand interbank loans");
+      });
+    });
+
+    it("Create an insurance company", () => {
+      createCompanyAndGetId("InstantlyHealth AG").then((companyId) => {
+        cy.visitAndCheckAppMount(`/companies/${companyId}/frameworks/eutaxonomy-financials/upload`);
+        cy.get("select[name=financialServicesTypes]").select("Insurance or Reinsurance");
+        cy.get("select[name=attestation]").select("Limited Assurance");
+        cy.get('input[name="reportingObligation"][value=Yes]').check();
+        cy.get("input[name=taxonomyEligibleNonLifeInsuranceActivities]").type("0.22");
+        submitEuTaxonomyFinancialsUploadFormAndGetDataId();
+        cy.visitAndCheckAppMount(`/companies/${companyId}/frameworks/eutaxonomy`);
+        cy.get('div[name="taxonomyEligibleNonLifeInsuranceActivities"]')
+          .should("contain", "Taxonomy-eligible non-life insurance economic activities")
+          .should("contain", "22");
+        cy.get("body").should("not.contain", "Trading portfolio");
+        cy.get("body").should("not.contain", "on demand interbank loans");
+      });
+    });
+
+    it("Create an Asset Manager", () => {
+      createCompanyAndGetId("Magic Super-Stoxx Holding").then((companyId) => {
+        cy.visitAndCheckAppMount(`/companies/${companyId}/frameworks/eutaxonomy-financials/upload`);
+        cy.get("select[name=financialServicesTypes]").select("Asset Management");
+        cy.get("select[name=attestation]").select("Limited Assurance");
+        cy.get('input[name="reportingObligation"][value=Yes]').check();
+        cy.get('div[name="AssetManagement"]').find('input[name="taxonomyEligibleActivity"]').type("0.23");
+        cy.get('div[name="AssetManagement"]').find('input[name="derivatives"]').type("0.24");
```

Comment:
Grundsätzlich bin ich kein Freund des hardcodens von Testdaten. Warum muss das hier gemacht werden und wird nicht über fixture Daten gelöst? Ist es weil hier bestimmte Typen von financials gebraucht werden und ich mir das suchen in den Daten ersparen will?

Reply:
Das ist ein Testfile, dass den gesamten User-Flow von Hochladen der Company über das UI, bis hin zum Anzeigen der Ergebnisse im UI abbildet, und prüft, ob alles Konsistent ist. Einen Test nach ähnlichem Muster gibt es auch non-financials

Reply:
Implementierung abgeschlossen